test/move.js

```diff
@@ -0,0 +1,19 @@
+var R = require('../source');
+var eq = require('./shared/eq');
+
+describe.only('move', function() {
+  it('should correctly moves an element from an index to another', function() {
+    eq(R.move(0, 1, [1,2,3]), [2,1,3]);
+    eq(R.move(2, 1, [1,2,3]), [1,3,2]);
+    eq(R.move(-1, 0, [1,2,3]), [3,1,2]);
+  });
+
+  it('should move the element at the end when the destination index is outside liste bounds', function() {
```

Comment:
But is this the correct behavior?  Why do we distinguish how it behaves with the source index and the destination one?

Reply:
Because of the flexibility of `move()` and more precisely `splice` and `slice`, we are able to make a list rotation, but it has some unexpected behaviour, like this one you pointed out. How do you think prevent this behaviour without breaking the rotation one?

Reply:
I'm not quite sure.  Perhaps we want to allow negative outbounds only down to the first element.  I really don't like this asymmetry:

```js
move(-50, 2, ['a', 'b', 'c', 'd', 'e', 'f']) //=> ['b', 'c', 'a', 'd', 'e', 'f']
move( 50, 2, ['a', 'b', 'c', 'd', 'e', 'f']) //=> ['a', 'b', 'c', 'd', 'e', 'f']
```

Reply:
Maybe these ones are enough:
```js
move(0, -1, ['a', 'b', 'c']) /// ['b', 'c', 'a']
move(-1, 0, ['a', 'b', 'c']) /// ['c', 'a', 'b']
```

Reply:
So? Should Ramda be more strict on indexes, or keep that asymmetric behaviour?

Reply:
I have long wished that we never started supporting negative indices.  They add complexity that is rarely warranted, even though they do slightly simplify the use in a few cases.  But given that we have them, perhaps we should use something conceptually like this for any index we work with:

```js
const realIndex = (index, {length}) => index < 0 
  ? max(length + index, 0)
  : min(length - 1, index) 
```

I think that's already how `slice` works -- just by defaulting to `Array.prototype` or `String.prototype`.  Could we do that here with both `from` and `to`?

That would yield:

```js
move( -2,  1, ['a', 'b', 'c', 'd', 'e', 'f']) //=> ['a', 'e', 'b', 'c', 'd', 'f']
move(  1, -2, ['a', 'b', 'c', 'd', 'e', 'f']) //=> ['a', 'c', 'd', 'e', 'b', 'f']
move(-50,  2, ['a', 'b', 'c', 'd', 'e', 'f']) //=> ['b', 'c', 'a', 'd', 'e', 'f']
move( 50,  2, ['a', 'b', 'c', 'd', 'e', 'f']) //=> ['a', 'b', 'f', 'c', 'd', 'e']
```
----------

Better to my mind, though, is never falling off the end like that.  If either of the indices (allowing for small negative values) is not present, then we return an unmodified copy of the list.

That might involve creating a `positiveFrom` like the `positiveTo` and, before the `splice`, adding

```js
if (positiveFrom < 0 || positiveFrom >= list.length ||
    positiveTo   < 0 || positiveTo   >= list.length) {
  return result;
}
```

yielding

```js
move( -2,  1, ['a', 'b', 'c', 'd', 'e', 'f']) //=> ['a', 'e', 'b', 'c', 'd', 'f']
move(  1, -2, ['a', 'b', 'c', 'd', 'e', 'f']) //=> ['a', 'c', 'd', 'e', 'b', 'f']
move(-50,  2, ['a', 'b', 'c', 'd', 'e', 'f']) //=> ['a', 'b', 'd', 'd', 'e', 'f']
move( 50,  2, ['a', 'b', 'c', 'd', 'e', 'f']) //=> ['a', 'b', 'c', 'd', 'e', 'f']
```
Does that seem correct?

Reply:
Do you want to do that update?  I'd be inclined to accept this with that change, although I would still like some feedback from @ramda/core .

Reply:
Updated. I didn't add the `realIndex` function in the `internal` directory, as I don't know if it should be added in a separate PR. Let me know if you want to see it appears in this PR.

Reply:
Well, I'm afraid that this code combines two of what I think of contradictory suggestions of mine.

I was suggesting something like this except with

```js
  var positiveFrom = from < 0 ? list.length + from : from;
  var positiveTo = to < 0 ? list.length + to : to;
```

so that  

```js
move( -2,  1, ['a', 'b', 'c', 'd', 'e', 'f']) //=> ['a', 'e', 'b', 'c', 'd', 'f']
move(  1, -2, ['a', 'b', 'c', 'd', 'e', 'f']) //=> ['a', 'c', 'd', 'e', 'b', 'f']
move(-50,  2, ['a', 'b', 'c', 'd', 'e', 'f']) //=> ['a', 'b', 'd', 'd', 'e', 'f']
move( 50,  2, ['a', 'b', 'c', 'd', 'e', 'f']) //=> ['a', 'b', 'c', 'd', 'e', 'f']
```

The idea is that a negative index only does something up to the length of the list.  After that, it becomes a no-op.  I think that stopping at the end of the array is far too surprising.

Reply:
Ok, I think there was a misunderstanding... I fixed it and I hope everything is ok now.